server/api/utils/builder.py

```diff
@@ -181,6 +182,15 @@ def make_kaniko_pod(
             else:
                 attr_value = default_service_account
 
+        if attribute == "node_selector":
```

Comment:
tbh I don't really like how this for loop is written. IMO `get_kaniko_spec_attributes_from_runtime` should be a dict of (attr, callback) and for some attrs it will be the identity callback and for others it will be what's inside the if section. If you think you understand me you can change it but if not I can do it in a follow up